### ISSUE TITLE
Rails 7.2 compatibility issue

### ISSUE THREAD

**Issue by chaadow** [ankane/lockbox#194]
Hi,

Rails 7.2 is about to get released with the creation of the [7-2 stable branch](https://github.com/rails/rails/tree/7-2-stable). As `main` now is mainly focused on Rails 8.

In october 2023, there was a refactor in Rails regarding how attributes were handled. Here is the [commit](https://github.com/rails/rails/commit/e0a55b038f7f2f50d1467876558be183be6cedaa#diff-2a6a5a31a6210a1e7036784fb31c738e1d2874bb5ae123d03737db02cb198876)

Unfortunately this breaks lockbox, since it uses this "private/internal" class attribute `attributes_to_define_after_schema_loads`,  that was removed in rails 7.2. and breaks the gem. Here is where it's used in [lockbox](https://github.com/ankane/lockbox/blob/82de74c11b662791de647ba323705c18df4e8cc2/lib/lockbox/model.rb#L298-L299)

**Comment by ankane** [comment]
Added support in the commit above.

**Comment by chaadow** [comment]
Thanks!!

**Comment by chaadow** [comment]
Hi @ankane I have to further analyze the commit, a bit hard for me. But somewhere in this patch, there is a SQL database call

which blocks my `bin/rails parallel:create|drop` because of 

```
ActiveRecord::NoDatabaseError: We could not find your database: app-test2. Available database configurations can be found in config/database.yml.
```

I'll let you know once i figure it out ( by opening another issue), but maybe this can help you a bit.

Thanks again

**Comment by ankane** [comment]
Fixed in 7fdcf49e2650118a56e41769cb9da412d2733b83
